apps/deploy-web/src/components/billing-usage/PaymentPopup/PaymentPopup.tsx

```diff
@@ -0,0 +1,324 @@
+
+  const [error, setError] = useState<string>();
+  const [errorAction, setErrorAction] = useState<string>();
+  const submittedAmountRef = useRef<string>("");
+  const { user } = d.useUser();
+  const {
+    confirmPayment: { isPending: isConfirmingPayment, mutateAsync: confirmPayment },
+    applyCoupon: { isPending: isApplyingCoupon, mutateAsync: applyCoupon }
+  } = d.usePaymentMutations();
+  const { pollForPayment, isPolling } = d.usePaymentPolling();
+
+  const paymentForm = d.useForm<z.infer<typeof paymentFormSchema>>({
+    defaultValues: {
+      amount: 0
+    },
+    resolver: d.zodResolver(paymentFormSchema)
+  });
+
+  const couponForm = d.useForm<z.infer<typeof couponFormSchema>>({
+    defaultValues: {
+      coupon: ""
+    },
+    resolver: d.zodResolver(couponFormSchema)
+  });
+
+  const threeDSecure = d.use3DSecure({
+    onSuccess: () => {
+      pollForPayment();
+      setShowPaymentSuccess({ amount: submittedAmountRef.current, show: true });
+      paymentForm.reset();
+      couponForm.reset();
+      onClose();
+    },
+    showSuccessMessage: false
+  });
+
+  const clearError = () => {
+    if (error) {
+      setError(undefined);
+      setErrorAction(undefined);
+    }
+  };
+
+  const onPayment = async ({ amount }: z.infer<typeof paymentFormSchema>) => {
+    if (!user?.id) {
+      console.error("Payment attempted without a user id");
+      setError("Unable to process payment. Please refresh the page and try again.");
+      return;
+    }
+
+    if (!selectedPaymentMethodId) {
+      return;
+    }
+
+    // Capture the submitted amount before starting the payment flow
+    submittedAmountRef.current = amount.toString();
+    clearError();
+
+    try {
+      const response = await confirmPayment({
+        userId: user.id,
+        paymentMethodId: selectedPaymentMethodId,
+        amount,
+        currency: "usd"
+      });
+
+      if (response && response.requiresAction && response.clientSecret && response.paymentIntentId) {
+        threeDSecure.start3DSecure({
+          clientSecret: response.clientSecret,
+          paymentIntentId: response.paymentIntentId,
+          paymentMethodId: selectedPaymentMethodId
+        });
+      } else if (response.success) {
+        pollForPayment();
+        setShowPaymentSuccess({ amount: submittedAmountRef.current, show: true });
+        paymentForm.reset();
+        couponForm.reset();
+        onClose();
+      } else {
+        throw new Error("Payment failed");
+      }
+    } catch (error: unknown) {
+      const errorInfo = d.handleStripeError(error);
+
+      setError(errorInfo.message);
+      setErrorAction(errorInfo.userAction);
+      enqueueSnackbar(<d.Snackbar title={errorInfo.message} iconVariant="error" />, { variant: "error" });
+    }
+  };
+
+  const onClaimCoupon = async ({ coupon }: z.infer<typeof couponFormSchema>) => {
+    try {
+      if (!user?.id) {
+        console.error("Coupon application attempted without a user id");
```

Comment:
comment: please log these in the error handler service which will log it into sentry